contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,817 @@
+    uint256 stakerTotalAmount = add(stakes[_pollId][msg.sender][_vote], amount);
+
+    require(
+      stakerTotalAmount <= checkReputation(_pollId, msg.sender, _key, _value, _branchMask, _siblings),
+      "voting-rep-insufficient-rep"
+    );
+    require(
+      stakerTotalAmount >= wmul(requiredStake, minStakeFraction),
+      "voting-rep-insufficient-stake"
+    );
+
+    colony.obligateStake(msg.sender, poll.domainId, amount);
+    colony.transferStake(_permissionDomainId, _childSkillIndex, address(this), msg.sender, poll.domainId, amount, address(this));
+
+    // Update the stake
+    poll.unpaidRewards = add(poll.unpaidRewards, amount);
+    poll.stakes[_vote] = add(poll.stakes[_vote], amount);
+    stakes[_pollId][msg.sender][_vote] = stakerTotalAmount;
+
+    // Increment counter & extend claim delay if staking for an expenditure state change
+    if (poll.stakes[YAY] == requiredStake && _vote == YAY && getSig(poll.action) == CHANGE_FUNCTION) {
+      bytes32 structHash = hashExpenditureStruct(poll.action);
+      expenditurePollCounts[structHash] = add(expenditurePollCounts[structHash], 1);
+      bytes memory claimDelayAction = createClaimDelayAction(poll.action, UINT256_MAX);
+      executeCall(_pollId, claimDelayAction);
+    }
+
+    // Claim tokens if fully staked
+    if (poll.stakes[YAY] == requiredStake && poll.stakes[NAY] == requiredStake) {
+      poll.events[STAKE_END] = now;
+      tokenLocking.claim(token, true);
+    }
+
+    emit PollStaked(_pollId, msg.sender, _vote, amount);
+  }
+
+  function unstakePoll(
+    uint256 _pollId,
+    uint256 _permissionDomainId, // For extension's arbitration permission
+    uint256 _childSkillIndex, // For extension's arbitration permission
+    uint256 _vote,
+    uint256 _amount
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Staking, "voting-rep-not-staking");
+    require(now < add(poll.events[CREATE], wmul(stakePeriod, STAKE_COMMIT_FRACTION)), "voting-rep-cannot-withdraw");
+
+    tokenLocking.transfer(token, _amount, msg.sender, true);
+
+    poll.unpaidRewards = sub(poll.unpaidRewards, _amount);
+    poll.stakes[_vote] = sub(poll.stakes[_vote], _amount);
+    stakes[_pollId][msg.sender][_vote] = sub(stakes[_pollId][msg.sender][_vote], _amount);
+
+    emit PollUnstaked(_pollId, msg.sender, _vote, _amount);
+  }
+
+  function submitVote(
+    uint256 _pollId,
+    bytes32 _voteSecret,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Submit, "voting-rep-poll-not-open");
+
+    uint256 userRep = checkReputation(_pollId, msg.sender, _key, _value, _branchMask, _siblings);
+
+    voteSecrets[msg.sender][_pollId] = _voteSecret;
+    poll.repSubmitted = add(poll.repSubmitted, userRep);
```

Comment:
What's stopping me calling this function multiple times, and having my reputation double-counted here?

Reply:
Whoops, didn't catch that. We added the `repSubmitted` counter only later.